src/dataframe/dataframe.jl

```diff
@@ -152,6 +152,10 @@ function DataFrame(column_eltypes::AbstractVector{T}, cnames::AbstractVector{Sym
                    categorical::Vector{Bool}, nrows::Integer) where T<:Type
     # upcast Vector{DataType} -> Vector{Type} which can hold CategoricalValues
     updated_types = convert(Vector{Type}, column_eltypes)
+    if length(categorical) != length(column_eltypes)
+        throw(ArgumentError("arguments column_eltypes and categorical must have the same length " *
```

Comment:
Isn't `DimensionMismatch` typically used rather than `ArgumentError` for cases such as this?

Reply:
right - I will fix it